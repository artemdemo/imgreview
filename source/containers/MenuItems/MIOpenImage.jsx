import React from 'react';
import OpenImageDialog from '../../components/OpenImageDialog/OpenImageDialog';
import Icon from '../../components/Icon/Icon';
import MainMenuItem from '../../components/MainMenu/MainMenuItem';

class MIOpenImage extends React.PureComponent {
    constructor(props) {
        super(props);

        this.openImgDialogRef = React.createRef();
    }

    onClick = () => {
        this.openImgDialogRef.current.openDialog();
    };

    render() {
        return (
            <React.Fragment>
                <MainMenuItem onClick={this.onClick}>
                    <Icon
                        name='folder-open-o'
                    />
                </MainMenuItem>
                <OpenImageDialog
                    ref={this.openImgDialogRef}
                />
            </React.Fragment>
        );
    }
}

export default MIOpenImage;
