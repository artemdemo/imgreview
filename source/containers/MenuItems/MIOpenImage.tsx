import React from 'react';
import OpenImageDialog from '../OpenImageDialog/OpenImageDialog';
import Icon from '../../components/Icon/Icon';
import MainMenuItem from '../../components/MainMenu/MainMenuItem';

type Props = {
    disabled: boolean;
};

type State = {
    open: boolean;
};

class MIOpenImage extends React.PureComponent<Props, State> {
    static readonly defaultProps = {
        disabled: false,
    };

    constructor(props) {
        super(props);

        this.state = {
            open: false,
        };
    }

    onClick = () => {
        this.setState({
            open: true,
        }, () => {
            requestAnimationFrame(() => {
                this.setState({
                    open: false,
                });
            });
        });
    };

    render() {
        const {disabled} = this.props;
        return (
            <React.Fragment>
                <MainMenuItem
                    onClick={this.onClick}
                    disabled={disabled}
                >
                    <Icon
                        name='folder-open-o'
                        title='Open'
                    />
                </MainMenuItem>
                <OpenImageDialog
                    open={this.state.open}
                />
            </React.Fragment>
        );
    }
}

export default MIOpenImage;
