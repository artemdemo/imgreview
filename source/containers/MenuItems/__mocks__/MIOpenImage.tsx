import React from 'react';

const MIOpenImage = props => (
    <div data-mock='MIOpenImage'>
        {props.children}
    </div>
);

export default MIOpenImage;
