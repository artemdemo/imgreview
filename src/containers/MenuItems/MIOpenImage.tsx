import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faFileImage } from '@fortawesome/free-regular-svg-icons';
import OpenImageDialog from '../OpenImageDialog/OpenImageDialog';
import TopMenuItem from '../../components/TopMenu/TopMenuItem';
import * as gaService from '../../services/ganalytics';
import { t } from '../../services/i18n';

type TProps = {
  disabled: boolean;
};

type TState = {
  open: boolean;
};

class MIOpenImage extends React.PureComponent<TProps, TState> {
  static readonly defaultProps = {
    disabled: false,
  };

  constructor(props) {
    super(props);

    this.state = {
      open: false,
    };
  }

  onClick = () => {
    this.setState(
      {
        open: true,
      },
      () => {
        requestAnimationFrame(() => {
          this.setState({
            open: false,
          });
        });
      }
    );

    gaService.sendEvent({
      eventCategory: gaService.EEventCategories.MenuClick,
      eventAction: gaService.EEventActions.OpenImage,
    });
  };

  render() {
    const { disabled } = this.props;
    return (
      <>
        <TopMenuItem
          onClick={this.onClick}
          disabled={disabled}
          title={t('menu.openImage')}
          stopPropagation={false}
        >
          <FontAwesomeIcon icon={faFileImage} />
        </TopMenuItem>
        <OpenImageDialog open={this.state.open} />
      </>
    );
  }
}

export default MIOpenImage;
