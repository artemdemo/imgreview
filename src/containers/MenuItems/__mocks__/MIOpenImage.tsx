import React from 'react';

const MIOpenImage = (props) => (
  <div data-mock="MIOpenImage">{JSON.stringify(props, null, 2)}</div>
);

export default MIOpenImage;
