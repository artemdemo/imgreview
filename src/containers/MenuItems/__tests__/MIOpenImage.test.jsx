import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import MIOpenImage from '../MIOpenImage';

jest.mock('../../../components/TopMenu/TopMenuItem');
jest.mock('../../../containers/OpenImageDialog/OpenImageDialog');

describe('MIOpenImage', () => {
    it('should render', () => {
        const tree = renderer.create(
            <MIOpenImage />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should handle open', () => {
        const wrapper = mount(
            <MIOpenImage />
        );
        wrapper.setState({
            open: true,
        });
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
