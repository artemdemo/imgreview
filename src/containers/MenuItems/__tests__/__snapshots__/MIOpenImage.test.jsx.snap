// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MIOpenImage should handle open 1`] = `
<MIOpenImage
  disabled={false}
>
  <TopMenuItem
    disabled={false}
    onClick={[Function]}
  >
    <div
      data-mock="TopMenuItem"
      onClick={[Function]}
    >
      <Icon
        name="folder-open-o"
        title="Open"
      >
        <div
          data-mock="Icon"
        />
      </Icon>
    </div>
  </TopMenuItem>
  <MIOpenImage
    open={true}
  >
    <div
      data-mock="OpenImageDialog"
    >
      open: 
      true
    </div>
  </MIOpenImage>
</MIOpenImage>
`;

exports[`MIOpenImage should render 1`] = `
Array [
  <div
    data-mock="TopMenuItem"
    onClick={[Function]}
  >
    <div
      data-mock="Icon"
    />
  </div>,
  <div
    data-mock="OpenImageDialog"
  >
    open: 
    false
  </div>,
]
`;
